prefect_azure/container_instance.py

```diff
@@ -470,7 +478,8 @@ def _configure_container(self) -> Container:
         ]
 
         # all container names in a resource group must be unique
-        container_name = str(uuid.uuid4())
+        number_suffix = ''.join(random.choices(string.digits, k=6))
+        container_name = self.name + '-' + number_suffix if self.name else str(uuid.uuid4())
```

Comment:
Azure's container instances service has some constraints on container names that we should respect. Also, to have a higher assurance of randomness, let's increase the suffix length and add letters.
```suggestion
        slugified_name = slugify(self.name, max_length=52, regex_pattern=r"[^a-zA-Z0-9-]+")
        random_suffix = ''.join(random.choices(string.ascii_lowercase, string.digits, k=10))
        container_name = self.name + '-' + number_suffix
```

`slugify` is a dependency of `prefect` so you should be able to import `slugify` with:
```
import slugify from slugify
```

Also, the `name` attribute for an infrastructure block is populated to the flow run name by default, so it should be populated in all cases.